### PR TITLE
ShapeFilter: syntax fix + more explicit error messages

### PR DESCRIPTION
**Pull request by dgirardeau**
I got a SHP file from a user with a really misleading error (about missing memory while it was actually a corrupted file). I fixed that, and the the syntax by the same way (to be more compliant with CC's code base ;).

### CODE REVIEW

File: libs/qCC_io/ShpFilter.cpp [hunk 1]
```diff
@@ -721,68 +701,74 @@ static std::vector<CCVector3> readPoints(QDataStream& shpStream, int32_t numPoin
...
 {
 	int32_t vertCount = lastIndex - firstIndex + 1;
+	if (vertCount <= 0)
+	{
+		assert(false);
+		return nullptr;
+	}
 
-	auto *vertices = new ccPointCloud("vertices");
+	ccPointCloud* vertices = new ccPointCloud("vertices");
```
Comment (asmaloney) [inline libs/qCC_io/ShpFilter.cpp:753]: I thought when we discussed `auto` way back in the C++11 change, this was acceptable?

Generally this is an acceptable "new C++ style" that [clang static analysis](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html) can warn about.

It avoids repeating the type that's already there and makes for easier-to-read code.

```cpp
auto foo = 5; //bad;
auto bar = new blat( "ding" ); //good
```

File: libs/qCC_io/ShpFilter.cpp [hunk 2]
```diff
@@ -794,30 +780,31 @@ ccMesh *createMesh(
...
+			areAllValuesOfPartsNaNs = std::all_of(scalarValues.begin() + firstIndex,
 			                                      scalarValues.begin() + lastIndex,
 			                                      [](ScalarType val)
 			                                      { return std::isnan(val); });
 		}
-		if (!areAllValuesOfPartsNans)
+		if (!areAllValuesOfPartsNaNs)
 		{
-			auto *sf = new ccScalarField("Measures");
+			ccScalarField* sf = new ccScalarField("Measures");
```
Comment (asmaloney) [inline libs/qCC_io/ShpFilter.cpp:801]: Same as above

File: libs/qCC_io/ShpFilter.cpp [hunk 3]
```diff
@@ -836,9 +823,10 @@ ccMesh *createMesh(
 }
 
 //! Builds the patches that where read from a MultiPatch record of a Shapefile
-//! patches are build as meshes, each part is its own mesh
-//! The only supported Patches/Part types are Triangle_Fan & Triangle_Strip
-CC_FILE_ERROR buildPatches(
+/** Patches are buit as meshes, each part is its own mesh.
+	The only supported Patches/Part types are Triangle_Fan & Triangle_Strip
+**/
```
Comment (asmaloney) [inline libs/qCC_io/ShpFilter.cpp:828]: Why switch to C-style comments?

Mixing them like this makes it more difficult to read.

File: libs/qCC_io/ShpFilter.cpp (hunk 1 above)
Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp:753]: I personally think it's less explicit (you have to look at the code on the right to understand the type. And it's particularly 'lazy' when it's not for very long templated types.

File: libs/qCC_io/ShpFilter.cpp (hunk 2 above)
Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp:801]: Same as above :D

File: libs/qCC_io/ShpFilter.cpp (hunk 3 above)
Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp:828]: I thought '//!' was for the brief description but apparently it's not... The issue is that it's like that everywhere in the code...

File: libs/qCC_io/ShpFilter.cpp (hunk 1 above)
Comment (asmaloney) [reply libs/qCC_io/ShpFilter.cpp:753]: You're never going to _only_ read `auto *vertices` or `ccPointCloud* vertices` and then stop. You're going to read the whole line anyway, so it's just extra noise.

Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp:753]: Not always. Depends on the circumstances. Really, for me auto is a shortcut that should only be used when necessary.

Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp:753]: I mean, if you only want to know the type, you'll get the information faster.

**Pull request merged**